src/lib/optimizer/strategy/dependent_group_by_reduction_rule.cpp

```diff
@@ -97,21 +99,25 @@ void DependentGroupByReductionRule::_apply_to_plan_without_subqueries(
     // Example: SELECT DISTINCT n_nationkey FROM nation is reflected as [ Aggregate GROUP BY n_nationkey ]. The
     // AggregateNode does not have more node expressions than the single GROUP BY column and the input (StoredTableNode
     // in this case) has a UCC for { n_nationkey }.
-    if (group_by_columns.size() == node->node_expressions.size() &&
-        node->left_input()->has_matching_ucc(group_by_columns)) {
-      const auto& output_expressions = aggregate_node.output_expressions();
-      // Remove the AggregateNode if it does not limit or reorder the output expressions.
-      if (expressions_equal(output_expressions, node->left_input()->output_expressions())) {
-        lqp_remove_node(node);
+    if (group_by_columns.size() == node->node_expressions.size()) {
+      auto matching_ucc = node->left_input()->get_matching_ucc(group_by_columns);
+      if (matching_ucc.has_value()) {
+        rule_was_applied_using_non_permanent_ucc = !matching_ucc->is_permanent();
```

Comment:
you assign this member inside `visit_lqp`. So when you rewrite an AggregateNode with "non-permanent" FD first and another one with "permanent" FD later, you have a wrong result. You should use `|=`, right?